demos/python_demos/rag_chatbot/servable_stream/model.py

```diff
@@ -0,0 +1,299 @@
+from langchain.docstore.document import Document
+
+class ChineseTextSplitter(CharacterTextSplitter):
+    def __init__(self, pdf: bool = False, **kwargs):
+        super().__init__(**kwargs)
+        self.pdf = pdf
+
+    def split_text(self, text: str) -> List[str]:
+        if self.pdf:
+            text = re.sub(r"\n{3,}", "\n", text)
+            text = text.replace("\n\n", "")
+        sent_sep_pattern = re.compile(
+            '([﹒﹔﹖﹗．。！？]["’”」』]{0,2}|(?=["‘“「『]{1,2}|$))')
+        sent_list = []
+        for ele in sent_sep_pattern.split(text):
+            if sent_sep_pattern.match(ele) and sent_list:
+                sent_list[-1] += ele
+            elif ele:
+                sent_list.append(ele)
+        return sent_list
+
+
+TEXT_SPLITERS = {
+    "Character": CharacterTextSplitter,
+    "RecursiveCharacter": RecursiveCharacterTextSplitter,
+    "Markdown": MarkdownTextSplitter,
+    "Chinese": ChineseTextSplitter,
+}
+
+
+LOADERS = {
+    ".csv": (CSVLoader, {}),
+    ".doc": (UnstructuredWordDocumentLoader, {}),
+    ".docx": (UnstructuredWordDocumentLoader, {}),
+    ".enex": (EverNoteLoader, {}),
+    ".epub": (UnstructuredEPubLoader, {}),
+    ".html": (UnstructuredHTMLLoader, {}),
+    ".md": (UnstructuredMarkdownLoader, {}),
+    ".odt": (UnstructuredODTLoader, {}),
+    ".pdf": (PDFMinerLoader, {}),
+    ".ppt": (UnstructuredPowerPointLoader, {}),
+    ".pptx": (UnstructuredPowerPointLoader, {}),
+    ".txt": (TextLoader, {"encoding": "utf8"}),
+}
+
+
+def load_single_document(file_path: str) -> List[Document]:
+    """
+    helper for loading a single document
+
+    Params:
+      file_path: document path
+    Returns:
+      documents loaded
+
+    """
+    ext = "." + file_path.rsplit(".", 1)[-1]
+    if ext in LOADERS:
+        loader_class, loader_args = LOADERS[ext]
+        loader = loader_class(file_path, **loader_args)
+        return loader.load()
+
+    raise ValueError(f"File does not exist '{ext}'")
+
+
+def default_partial_text_processor(partial_text: str, new_text: str):
+    """
+    helper for updating partially generated answer, used by default
+
+    Params:
+      partial_text: text buffer for storing previosly generated text
+      new_text: text update for the current step
+    Returns:
+      updated text string
+
+    """
+    partial_text += new_text
+    return partial_text
+
+
+text_processor = llm_model_configuration.get(
+    "partial_text_processor", default_partial_text_processor
+)
+
+
+def deserialize_prompts(batch_size, input_tensor):
+    if batch_size == 1:
+        return [bytes(input_tensor).decode()]
+    np_arr = deserialize_bytes_tensor(bytes(input_tensor))
+    return [arr.decode() for arr in np_arr]
+
+
+def serialize_completions(batch_size, result):
+    if batch_size == 1:
+        return [Tensor("completion", result.encode())]
+    return [Tensor("completion", serialize_byte_tensor(
+        np.array(result, dtype=np.object_)).item())]
+
+
+
+class OvmsPythonModel:
```

Comment:
Until we don't have a better solution, please print with `flush=True`

Reply:
Done